Update calendar.php (full new version).

phili67 opened the pull request:
This is new calendar, it will only work with :
- api/routes/events.php
- Calendrier.php
- ChurchCRM/services/CalendarService.php
- ChurchCRM/model/Event.php
- ChurchCRM/model/Person.php
- api/routes/events.php

All the settings with the buttons are stored, so when you revisit the page, you will retrieve your customization.

So now it's possible to create events in the calendar, to move them, to resize them etc ...
![capture d ecran 2017-11-13 a 22 29 17](https://user-images.githubusercontent.com/20263693/32750250-62c741ca-c8c2-11e7-8951-3bf06ffa42be.png)

You can see, that the events are colorized in function of a new function I wrote inside the code.

I put the age near the name, for the birthdate.

It's possibile to filter them by groups, the filter by type is not yet done.
![capture d ecran 2017-11-13 a 22 29 50](https://user-images.githubusercontent.com/20263693/32750307-8bab0df6-c8c2-11e7-891d-5eeaaa678f54.png)

When you select a range this window appear :
![capture d ecran 2017-11-13 a 22 29 26](https://user-images.githubusercontent.com/20263693/32750261-6e663702-c8c2-11e7-8eea-1f32ac62f555.png)

You cans see that the date are in function of the hours/minutes etc ....
![capture d ecran 2017-11-13 a 22 29 39](https://user-images.githubusercontent.com/20263693/32750270-744ea0be-c8c2-11e7-9a4f-c8cfac5b2380.png)

You can show/hide birthday :
![capture d ecran 2017-11-13 a 22 33 50](https://user-images.githubusercontent.com/20263693/32750393-d108c8fc-c8c2-11e7-9bd3-9cd17c605d4a.png)

For our calendar, there's so many birthday, I use the option withLimit, which will reduce the number of item in a cell.
![capture d ecran 2017-11-13 a 22 35 31](https://user-images.githubusercontent.com/20263693/32750456-02544ada-c8c3-11e7-85d3-a178050a60db.png)

I'm really proud of the result.

I will add the way to restrict the edition, when a user don't have the right to edit events, in function of the fact he's in a group or not.

The goal is to hide the groups the user don't belong too.

Last When a user log, I plan to build a personal environnement :
- personal data
- personal calendar
- I dream about the way that the event can send directly to the attached people.

phili67 commented:
I've to wait to the #3293 upgrade.

phili67 commented:
The goal when I build the new event is to do this per default the event is personnal with no group assigned, this is working, I'm waiting of the new EventQuery and new Event class.
![capture d ecran 2017-11-14 a 08 37 37](https://user-images.githubusercontent.com/20263693/32768195-27ae8b48-c917-11e7-981e-c557b78443e2.png)

It's possibile to add the add a group, when you add one, the window toogle, like this :
![capture d ecran 2017-11-14 a 08 37 43](https://user-images.githubusercontent.com/20263693/32768239-4b2928d0-c917-11e7-8f8f-a24ddba1bde7.png)

If an event doesn't have a group assigned it's an person_event.

phili67 commented:
@DawoudIO I really want that you never use this code in ChurchCRM, this is my own code.
Thank you

DawoudIO commented:
testing now

DawoudIO commented:
I really like it... a few comments... 
- the value in the boxes kinda get in the way where you need to delete before you enter the text
- after you add an event there is a dbl copy shows for a split sec then it goes away

DawoudIO commented on src/ChurchCRM/model/ChurchCRM/Person.php:
```diff
@@ -413,4 +413,36 @@ public function postSave(ConnectionInterface $con = null) {
       return parent::postSave($con);
     }
+    
+    public function getAge()
```
we did this before via https://momentjs.com/

phili67 replied:
For me when the age is hidden, the birthdate should be stay in the background in php.

DawoudIO commented on src/api/routes/events.php:
```diff
@@ -12,15 +18,133 @@
+       $event->setEnd($newEnd->format('Y-m-d H:i:s'));
+       $event->save();
+  
+        $calendarS = new CalendarService();
```
you don't need 2 of the same service

DawoudIO replied:
same as above

DawoudIO commented on src/api/routes/events.php:
```diff
@@ -12,15 +18,133 @@
+           $eventCount->save();
+         }
+     
+         $calendarS = new CalendarService();
```
use $this->CalendarService it comes from  https://github.com/ChurchCRM/CRM/blob/master/src/api/dependencies.php

DawoudIO commented on src/api/routes/events.php:
```diff
@@ -12,15 +18,133 @@
+       $event->setEnd(str_replace("T"," ",$input->end));
+       $event->save();
+  
+        $calendarS = new CalendarService();
```
same

DawoudIO commented on src/api/routes/events.php:
```diff
@@ -12,15 +18,133 @@
+         $event->save(); 
+     
+         if ($input->Total > 0 || $input->Visitors || $input->Members){
+           $eventCount = new EventCounts; 
```
you can create a Contractor that takes the params or a method that removes the copy/past

phili67 replied:
I don't understand you ?

DawoudIO commented on src/api/routes/events.php:
```diff
@@ -12,15 +18,133 @@
     
-    
- 
+        $calendarS = new CalendarService();
```
same

DawoudIO commented on src/api/routes/events.php:
```diff
@@ -12,15 +18,133 @@
+     }
+     
 
+     //return $response->write($event->toJSON());
```
del

DawoudIO commented on src/calendar.php:
```diff
@@ -51,27 +107,396 @@
 <!-- fullCalendar 2.2.5 -->
 <script>
+
+  var anniversary = false;
```
why off by default

phili67 replied:
Cause we have : 1500 users ;-).
I can't support to see them all ....
It's more easy to read.

DawoudIO commented on src/calendar.php:
```diff
@@ -51,27 +107,396 @@
+  var withlimit   = false;
+  
+ 
+  var birthD = localStorage.getItem("birthday");
```
all these JS should be moved into /js/calendar.js to clean up the page

phili67 replied:
Ok I know, but it's still under development, but it's fully functional.
Last there's a lot of php code inside to create the menu in function of the group etc ...
We cans leave the code as this for the first version.

DawoudIO replied:
yes can you create a issue to track the extract to js file

phili67 commented:
I've added  :
- when you click the text it will disappear "Calendar Title" 
- and Calendar description
- I've added too a dialog with timeout which said "Event was added successfully."
![capture d ecran 2017-11-16 a 20 56 28](https://user-images.githubusercontent.com/20263693/32912862-b50cf402-cb10-11e7-8947-3369546870cc.png)

phili67 commented:
I think it's now OK though I thought we could add the first stone to a personal environnement too with the tables I sent you

phili67 commented:
@DawoudIO I've changed what you want

DawoudIO commented on src/api/routes/events.php:
```diff
@@ -12,15 +29,130 @@
+       $event->setEnd($newEnd->format('Y-m-d H:i:s'));
+       $event->save();
+  
+        $calendarS = $this->CalendarService;
```
you don't need to create a var, just use it as is

crossan007 commented on src/calendar.php:
```diff
@@ -35,6 +60,49 @@
+  <div class="box-body">
+      <form>
+          <div class="col-sm-3"> <b><?= gettext("Birthdate") ?>:</b> <input data-size="small" id="isBirthdateActive" type="checkbox" data-toggle="toggle" data-on="<?= gettext("Include") ?>" data-off="<?= gettext("Exclude") ?>"> </div>
+          <div class="col-sm-3"> <b><?= gettext("anniversary") ?>:</b> <input data-size="small" id="isAnniversaryActive" type="checkbox" data-toggle="toggle" data-on="<?= gettext("Include") ?>" data-off="<?= gettext("Exclude") ?>"></div>
```
anniversary should be "Anniversary"

phili67 commented:
@crossan007 @DawoudIO  I've just update the code, when a user don't have the right bAddEvent, He can't add event in the calendar. There is a bug in the all the Header-Function new and old.
Normally when bAddEvent is fixed for the user the add menu should disappear.
![capture d ecran 2017-11-17 a 08 52 11](https://user-images.githubusercontent.com/20263693/32936493-c24d6930-cb74-11e7-94b9-a04b28dfb577.png)


@crossan007 Last, It's fine to fix some spelling problems.
When I test the code,  the drag and drop was no more functional ?

Please not reproducing the problem of the PhotoBook.

It's nice to do this, but I've tested the code in every direction, I follow your way for your foreach and add change the spelling problem.

So I correct it again.

I think when the code is at the end to be merge, we don't have to modify something, send me an email or something else.

phili67 commented:
I think, we've to do the same thing in events.php
It should be a hack to inject code inside.
Should I add this last part ?
Are we near to put out the 2.9.0.

DawoudIO commented:
we have a test that fails 

    features/Groups/GroupsList.feature:24

phili67 commented:
apply style CI

DawoudIO commented:
waiting for a test pass.
